site/components/src/main/java/com/visitscotland/brxm/rest/event/HstQueryService.java

```diff
@@ -0,0 +1,41 @@
+package com.visitscotland.brxm.rest.event;
+
+import org.hippoecm.hst.container.RequestContextProvider;
+import org.hippoecm.hst.content.beans.query.HstQuery;
+import org.hippoecm.hst.content.beans.query.HstQueryResult;
+import org.hippoecm.hst.content.beans.query.builder.HstQueryBuilder;
+import org.hippoecm.hst.content.beans.query.exceptions.QueryException;
+import org.hippoecm.hst.content.beans.standard.HippoBean;
+import org.hippoecm.hst.core.component.HstComponentException;
+import org.hippoecm.hst.core.component.HstRequest;
+import org.springframework.stereotype.Component;
+
+import javax.jcr.*;
+
+@Component
+public class HstQueryService {
+
+    //TODO: Create CMS configuration property
+    private static final String EVENTS_LOCATION = "/content/documents/bsh/sandbox/events";
+
+    public <T extends HippoBean> HstQueryResult  query(Class<T> documentType) {
```

Comment:
Small point, when I am working with generics I aim to keep the signatures consistent, such as:

```suggestion
    public <H extends HippoBean> HstQueryResult query(Class<H> hippoBean) {
```

if `documentType` is important here, let's add a JavaDoc to explain the purpose of the method or lock down the generics as much as we can to prevent receiving problematic types. There should also be validation done here too in my eyes. What do you think?